olive/passes/onnx/moe_experts_distributor.py

```diff
@@ -259,12 +368,38 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
         config.update(get_external_data_config())
         return config
 
+    @staticmethod
+    def _validate_world_size(v):
+        if int(v) < 2:
+            raise ValueError("world_size should be >= 2")
+
+        return v
+
     @staticmethod
     def _validators() -> Dict[str, Callable]:
-        return {"validate_distributor_config": validator("world_size", allow_reuse=True)(_validate_world_size)}
+        return {
+            "validate_distributor_config": validator("world_size", allow_reuse=True)(
+                MoEExpertsDistributor._validate_world_size
+            )
+        }
 
     def _run_for_config(
         self, model: ONNXModel, data_root: str, config: Dict[str, Any], output_model_path: str
     ) -> DistributedOnnxModel:
-        output_filepaths = run(config["world_size"], model.model_path, output_model_path)
-        return DistributedOnnxModel(output_filepaths)
+        # huggingface/tokenizers: The current process just got forked, after parallelism has already been used.
+        # Disabling parallelism to avoid deadlocks...
+        # To disable this warning, you can either:
+        #     - Avoid using `tokenizers` before the fork if possible
+        #     - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
+        os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+        strategy = MoEExpertDistributionPatternMatcherA(config["world_size"], model.model_path)
+        experts, num_experts = strategy.identify_experts(output_model_path)
+        output_filepaths = strategy.distribute(
+            experts,
```

Comment:
I'd avoid using the word 'strategy' here so that folks are not confused with usual model distribution strategy such as Expert Parallel, Pipeline Parallel, Tensor Parallel etc.,

Reply:
Replaced 'strategy' with 'matcher'.